unitTests/utils/UrlUtilsTest.ts

```diff
@@ -266,5 +266,57 @@ export function UrlUtilsTest() {
       });
       expect(url).toBe(`https://a.com?123=4&abc=d&efg=h`);
     });
+
+    describe('getLinkDestination', () => {
```

Comment:
I think this may be one of the most important parts to review in this PR. This defines the intended behaviour of `getLinkDestination`.

Reply:
Although, all of this should not be handled by the UI :) 

So, it kinda sucks to say this, and I'm sorry in advance :D But this should all be removed.

The ML model should not be sending us something that is not usable, and that we have to do tons of hackish manipulation. This is the very definition of a flaky API, and not something we should accept, as it also means this API is un-usable by our clients directly.